nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/GetTCP.java

```diff
@@ -0,0 +1,668 @@
+            final String[] hostPortPairs = value.split(",");
+            boolean validHostPortPairs = true;
+            String reason = "";
+            String offendingSubject = subject;
+
+            if(0 == hostPortPairs.length){
+                return new ValidationResult.Builder().subject(subject).input(value).valid(false).explanation(offendingSubject + " cannot be empty").build();
+            }
+            for (final String hostPortPair : hostPortPairs) {
+                offendingSubject = hostPortPair;
+                //split pair
+                if (hostPortPair.isEmpty()) {
+                    validHostPortPairs = false;
+                    reason = "endpoint is empty";
+                    break;
+                }
+                if (!hostPortPair.contains(":")) {
+                    validHostPortPairs = false;
+                    reason = "endpoint pair does not contain valid delimiter";
+                    break;
+                }
+                final String[] parts = hostPortPair.split(":");
+
+                if (1 == parts.length) {
+                    validHostPortPairs = false;
+                    reason = "could not determine the port";
+                    break;
+                } else {
+                    try {
+                        final int intVal = Integer.parseInt(parts[1]);
+
+                        if (intVal <= 0) {
+                            reason = "not a positive value";
+                            validHostPortPairs = false;
+                            break;
+                        }
+                    } catch (final NumberFormatException e) {
+                        reason = "not a valid integer";
+                        validHostPortPairs = false;
+                        break;
+                    }
+                }
+                //if we already have a bad pair then exit now.
+                if (!validHostPortPairs) {
+                    break;
+                }
+            }
+            return new ValidationResult.Builder().subject(offendingSubject).input(value).explanation(reason).valid(validHostPortPairs).build();
+        }
+    };
+
+    public static final PropertyDescriptor ENDPOINT_LIST = new PropertyDescriptor
+            .Builder().name("Endpoint List")
+            .description("A comma delimited list of the servers to connect to. The format should be " +
+                    "<server_address>:<port>. Only one server will be connected to at a time, the others " +
+                    "will be used as fail overs.")
+            .required(true)
+            .addValidator(ENDPOINT_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor FAILOVER_ENDPOINT = new PropertyDescriptor
+            .Builder().name("Failover Endpoint")
+            .description("A failover server to connect to if one of the main ones is unreachable after the connection " +
+                    "attempt count. The format should be <server_address>:<port>.")
```

Comment:
There are actually two different use case in play. That first is that there is a single server you want to connect to and get data from and if that fails then you want to connect to a backup server. I can see how the endpoint list maybe used to satisfy this. However, there is a second use case which may not be apparent. In the second case the endpoint list forms a logical grouping of servers to connect to simultaneously and receive information. Thus, they are all active and do not serve as a backup. In this case the failover server is used when connectivity to one of the endpoint servers fails.

I am totally open to other ways of handling the two use cases if this is confusing/misleading/or not as clean as it could be.